R/QFeatures-class.R

```diff
@@ -618,8 +641,8 @@ setReplaceMethod("names", c("QFeatures", "character"),
                      for (i in seq_len(length(x))) {
                          al <- x@assayLinks[[i]]
                          al@name  <- unname(key_vals[key_vals[, 1] == al@name, 2])
-                         if (!is.na(al@from))
-                             al@from <- unname(key_vals[key_vals[, 1] == al@from, 2])
+                         if (!all(is.na(al@from)))
+                             al@from <- unname(key_vals[key_vals[, 1] %in% al@from, 2])
```

Comment:
I see you changed `==` into `%in%` here. Does that mean that the order isn't important?

Reply:
Ouch I can't remember why I changed this... I'll revert this change.

Reply:
I couldn't say without more in-depth inspection, so better double check.

Reply:
Ok I remember. When an assay as multiple elements in `from`, ie it has multiple parent assays, a warning is thrown:
```r
In key_vals[, 1] == al@from :
  longer object length is not a multiple of shorter object length
```
I'll include a unit test for this so that I don't forget next time :sweat_smile: